types/micro/index.d.ts

```diff
@@ -10,7 +10,7 @@ export type RequestHandler = (req: IncomingMessage, res: ServerResponse) => any;
 
 export function run(req: IncomingMessage, res: ServerResponse, fn: RequestHandler): Promise<void>;
 
-declare function serve(fn: RequestHandler): Server;
+declare function serve(fn: RequestHandler | Promise<RequestHandler>): Server;
```

Comment:
This is incorrect. According to the document, the _result_ of `RequestHandler` can be a promise, not the handler itself:

```js
micro(Promise.resolve((req, res) => {})); // error
micro(async (req, res) => {})); // ok
```

Since `RequestHandler` is already defined to return `any`, you can already return a promise should you so choose.

Reply:
oh yes thanks for having a second look